solver/src/driver.rs

```diff
@@ -230,8 +231,8 @@ impl Driver {
             .into_iter()
             .zip(simulations)
             .partition_map(|(settlement, result)| match result {
-                Ok(()) => Either::Left(settlement),
-                Err(err) => Either::Right((settlement, err)),
+                Ok(_) => Either::Left(settlement),
```

Comment:
One advantage of `Ok(())` is that if the type of the `result` changes, you would get a compiler error to update at this location. Very minor though, and I also see the argument that `Ok(_)` could more clearly communicate the intent of "it succeeded, don't care about the result".

Not requesting any changes, just rambling.

Reply:
This is exactly why it's _ now. The type has changed from () to U256 for the gas estimation. Next PR will make use of the value.